varint.go

```diff
@@ -76,18 +76,18 @@ func ReadUvarint(r io.ByteReader) (uint64, error) {
 	// released under the BSD License.
 	var x uint64
 	var s uint
-	for i := 0; ; i++ {
+	for s = 0; ; s += 7 {
 		b, err := r.ReadByte()
 		if err != nil {
-			if err == io.EOF && i != 0 {
+			if err == io.EOF && s != 0 {
 				// "eof" will look like a success.
 				// If we've read part of a value, this is not a
 				// success.
 				err = io.ErrUnexpectedEOF
 			}
 			return 0, err
 		}
-		if (i == 8 && b >= 0x80) || i >= MaxLenUvarint63 {
+		if (s == 56 && b >= 0x80) || s >= (7*MaxLenUvarint63) {
```

Comment:
this is the only use of `MaxLenUvarint63` - should we have the constant be 63, rather than a 7 here and 9 as the constant?

Reply:
It's a public constant and useful for checking buffer sizes.

Reply:
(also used by go-msgio, which we really need to kill)